src/gazebo_mavlink_interface.cpp

```diff
@@ -594,10 +584,7 @@ void GazeboMavlinkInterface::OnUpdate(const common::UpdateInfo&  /*_info*/) {
     last_heartbeat_sent_time_ = current_time;
   }
 
-  // Always send Gyro and Accel data at full rate (= sim update rate)
-  SendSensorMessages();
```

Comment:
The imu message is used as a signal for lockstep: https://docs.px4.io/main/en/simulation/#simulator-mavlink-api

Therefore if the sensor message needs to be sent when the simulation is updated, not whenever there is an imu message received from the plugin.

Reply:
Ah thanks @Jaeyoung-Lim , i missed that. I wanted to put multiple IMUs in the Gazebo simulation at different locations on the vehicle, which doesn't need to be on the same rigid body. So i tried to handle the IMU message the same way as e.g. the GPS message, where we can have multiple instances.

Reply:
@KonradRudin This should be already kind of supported on the simulation side. What is missing is on the px4 side which should only progress the lockstep for a certain instance of the imu message, while currently it does it for all imu messages

Reply:
Ignorant hot take -- does it make sense to have the lockstep fixed on a sensor message? Or should we just have a defined update on the simulation side? (I.e. was there a particularly good reason to use IMU for this? or was it just the shortest path solution when lockstep was added, and IMU was the highest frequency sensor?)

Reply:
@tstastny The reason is because gazebo doesn't really support an external lockstep interface and we have to block the physics update. I think you are probably right with the groundtruth message, but this is probably an afterthought of trying to do something that the architecture does not support. It was probably because the real information we actually care from a physics update is the sensor and not the dynamics update. Initially the `HIL_SENSOR` message was synchronized and therefore all sensors were being updated at once. This has evolved into the current multi instance `HIL_SENSOR` message and due to the corner cases we are hitting like this, we are trying to move towards ignition gazebo which supports a distributed lockstep interface https://github.com/PX4/PX4-Autopilot/pull/20057

Given that said, whether we use the groundtruth or the imu message as a lockstep signal, we still need to handle the multi-instance lockstep signal on the px4side. This shouldn't be too hard since we just need to check the instance id of the imu message and only progress lockstep on that side